nucypher/characters/lawful.py

```diff
@@ -942,6 +951,25 @@ def __init__(self,
             message = "Initialized Stranger {} | {}".format(self.__class__.__name__, self)
             self.log.debug(message)
 
+    def run(self,
+            learning: bool = True,
+            availability: bool = True,
+            worker: bool = True):
+
+        """Schedule Async Loops"""
+
+        if learning:
+            self.start_learning_loop(now=True)
+        if availability:
+            self._availability_sensor.start(now=False)  # wait...
+        if worker and not self.federated_only:
+            self.work_tracker.start(act_now=True)
```

Comment:
Is it possible for the work tracker to actually check what the availability sensor shows (if the availability sensor exists at all), and confirm activity only if the sensor shows that everything is fine?

Reply:
Okay, as a start - I pushed an update that allows for the worker to accept a confirmation requirement condition based on the sensor's status.